lib/plivo/xml/multipartycall.rb

```diff
@@ -0,0 +1,175 @@
+module Plivo
+  module XML
+    class MultiPartyCall < Element
+      @nestables = []
+      @valid_attributes = %w[role maxDuration maxParticipants waitMusicUrl
+                             waitMusicMethod agentHoldMusicUrl agentHoldMusicMethod
+                             customerHoldMusicUrl customerHoldMusicMethod record
+                             recordFileFormat recordingCallbackUrl recordingCallbackMethod
+                             statusCallbackEvents statusCallbackUrl statusCallbackMethod
+                             stayAlone coachMode mute hold startMpcOnEnter endMpcOnExit
+                             enterSound enterSoundMethod exitSound exitSoundMethod
+                             onExitActionUrl onExitActionMethod relayDTMFInputs]
+
+      VALID_ROLE_VALUES = %w[agent supervisor customer]
+      VALID_MAX_DURATION_VALUES = (300..28800).to_a
+      VALID_MAX_PARTICIPANT_VALUES = (2..10).to_a
+      VALID_WAIT_MUSIC_METHOD_VALUES = %w[GET POST]
+      VALID_AGENT_HOLD_MUSIC_METHOD_VALUES = %w[GET POST]
+      VALID_CUSTOMER_HOLD_MUSIC_METHOD_VALUES = %w[GET POST]
+      VALID_RECORD_VALUES = [true, false]
+      VALID_RECORD_FILE_FORMAT_VALUES = %w[mp3 wav]
+      VALID_RECORDING_CALLBACK_METHOD_VALUES = %w[GET POST]
+      VALID_STATUS_CALLBACK_METHOD_VALUES = %w[GET POST]
+      VALID_STAY_ALONE_VALUES = [true, false]
+      VALID_COACH_MODE_VALUES = [true, false]
+      VALID_MUTE_VALUES = [true, false]
+      VALID_HOLD_VALUES = [true, false]
+      VALID_START_MPC_ON_ENTER_VALUES = [true, false ]
+      VALID_END_MPC_ON_EXIT_VALUES = [true, false ]
+      VALID_ENTER_SOUND_METHOD_VALUES = %w[GET POST]
+      VALID_EXIT_SOUND_METHOD_VALUES = %w[GET POST]
+      VALID_ON_EXIT_ACTION_METHOD_VALUES = %w[GET POST]
+      VALID_RELAY_DTMF_INPUTS_VALUES = [true, false ]
+      
+
+      def initialize(body, attributes = {})
+        if attributes[:role] && !VALID_ROLE_VALUES.include?(attributes[:role].downcase)
+          raise PlivoXMLError, "invalid attribute value #{attributes[:role]} for maxDuration"
+        elsif !attributes[:role]
+          raise PlivoXMLError, "role not mentioned : possible values - Agent / Supervisor / Customer"
+        end
+
+        if attributes[:maxDuration] && !VALID_MAX_DURATION_VALUES.include?(attributes[:maxDuration])
+          raise PlivoXMLError, "invalid attribute value #{attributes[:maxDuration]} for maxDuration"
+        elsif !attributes[:maxDuration]
+          attributes[:maxDuration] = 14400
+        end
+
+        if attributes[:maxParticipants] && !VALID_MAX_PARTICIPANT_VALUES.include?(attributes[:maxParticipants])
+          raise PlivoXMLError, "invalid attribute value #{attributes[:maxParticipants]} for maxParticipants"
+        elsif !attributes[:maxParticipants]
+          attributes[:maxParticipants] = 10
+        end
+
+        if attributes[:waitMusicMethod] && !VALID_WAIT_MUSIC_METHOD_VALUES.include?(attributes[:waitMusicMethod])
```

Comment:
make all method checks case insensitive using `downcase`

Reply:
@kunal-plivo is it `uppercase` or `downcase` for the method checks ?  The possible values come under `%w[GET POST]`

Reply:
@Koushik-Ayila `uppercase` it would be